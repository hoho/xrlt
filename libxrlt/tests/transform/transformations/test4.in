id:0, type:100, last:0, error:0, data:
id:0, type:100, last:0, error:0, data:
id:1, type:400, last:0, error:0, data:200
id:1, type:600, last:1, error:0, data:{"data": [{"ololo": "yep", "yoyo": ["a", "b", "c"]}, {"alala": "yep", "yoyo": [{"a": "aaa"}, {"b": "bbb"}, {"c": "ccc"}]}]}
id:0, type:100, last:0, error:0, data:
