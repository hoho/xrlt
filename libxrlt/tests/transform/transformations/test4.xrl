<xrl:requestsheet xmlns:xrl="http://xrlt.net/Transform">

    <xrl:transformation name="json-stringify" type="json-stringify" />
    <xrl:transformation name="xslt-stringify" type="xslt-stringify" src="test4.xsl" />

    <xrl:response>
        <xrl:include>
            <xrl:href>
                <xrl:text>/tmp/tmp</xrl:text>
            </xrl:href>
            <xrl:type>json</xrl:type>

            <xrl:with-param name="q">
                <xrl:transform name="json-stringify">
                    <tmp1>Blah blah</tmp1>
                    <tmp2>Mooo mooo</tmp2>
                </xrl:transform>
            </xrl:with-param>

            <xrl:success>
                <xrl:transform name="json-stringify">
                    <tmp3>
                        <xrl:transform name="xslt-stringify">
                            <tmp4>
                                <xrl:for-each select="/data[ololo]/yoyo">
                                    <item>
                                        <xrl:copy-of select="node()" />
                                    </item>
                                </xrl:for-each>
                            </tmp4>
                            <tmp5>
                                <xrl:for-each select="/data[alala]/yoyo">
                                    <item2>
                                        <xrl:copy-of select="*" />
                                    </item2>
                                </xrl:for-each>
                            </tmp5>
                        </xrl:transform>
                    </tmp3>
                </xrl:transform>
            </xrl:success>
        </xrl:include>
    </xrl:response>

</xrl:requestsheet>
