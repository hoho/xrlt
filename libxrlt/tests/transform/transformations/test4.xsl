<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">

    <xsl:output method="html" />

    <xsl:key name="kkk" match="/tmp5/item2/*/text()" use="local-name(..)" />

    <xsl:template match="/">
        <xsl:apply-templates select="/tmp4/item" />
    </xsl:template>

    <xsl:template match="item">
        <tmp key="{text()}"><xsl:value-of select="key('kkk', text())" /></tmp>
    </xsl:template>

</xsl:stylesheet>
