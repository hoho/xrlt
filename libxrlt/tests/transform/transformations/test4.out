XRLT_STATUS_SUBREQUEST
sr id: 1
sr method: GET
sr type: JSON
sr url: /tmp/tmp
sr query: q=%7B%22tmp1%22%3A+%22Blah+blah%22%2C+%22tmp2%22%3A+%22Mooo+mooo%22%7D
sr body: (null)
XRLT_STATUS_WAITING
XRLT_STATUS_WAITING
XRLT_STATUS_CHUNK
chunk: {"tmp3": "<tmp key=\"a\">aaa</tmp><tmp key=\"b\">bbb</tmp><tmp key=\"c\">ccc</tmp>\n"}
XRLT_STATUS_DONE
